inv/inventory.go

```diff
@@ -279,6 +281,55 @@ func (i *inventory) checkAttributesLimits(
 	return nil
 }
 
+const oneDay = 24 * time.Hour
+
+func (i *inventory) needsUpsert(
+	device *model.Device,
+	upsertAttrs model.DeviceAttributes,
+	removeAttrs model.DeviceAttributes,
+) bool {
+	needsUpsert := true
+	if device != nil {
+		// we update the inventory attributes at least once every (calendar) day
+		if !device.UpdatedTs.IsZero() &&
+			device.UpdatedTs.Truncate(oneDay) != time.Now().Truncate(oneDay) {
+			return true
+		}
+		needsUpsert = false
+		for _, attribute := range upsertAttrs {
+			if attribute.Scope != model.AttrScopeInventory {
+				needsUpsert = true
+				break
+			}
+			attributeChanged := true
+			for _, deviceAttribute := range device.Attributes {
+				if !(attribute.Scope == deviceAttribute.Scope &&
+					attribute.Name == deviceAttribute.Name) {
+					continue
+				}
+				attributeChanged = attribute.Value != deviceAttribute.Value
```

Comment:
This will return true on every slice, I'd suggest changing it to:
```suggestion
				attributeChanged = !reflect.DeepEqual(attribute.Value, deviceAttribute.Value)
```